docker/Dockerfile.builder

```
@@ -0,0 +1,9 @@
+# One-time builder image with Rust toolchain + all system deps pre-installed.
+# Built once, then reused for fast incremental compiles via bind mounts.
+FROM rust:1.86-slim
+
+RUN apt-get update && apt-get install -y --no-install-recommends \
```

Hadolint (2.14.0)

[warning] Pin versions in apt get install. Instead of `apt-get install <package>` use `apt-get install <package>=<version>`

(DL3008)